components/availability/src/Availability.svelte

```diff
@@ -337,27 +344,33 @@
         let availability = AvailabilityStatus.FREE; // default
         if (allCalendars.length) {
           for (const calendar of allCalendars) {
-            let availabilityExistsInSlot = calendar.timeslots.some((block) => {
-              if (calendar.availability === AvailabilityStatus.FREE) {
-                // typical use case
-                return time < block.end_time && block.start_time < endTime;
-              } else if (calendar.availability === AvailabilityStatus.BUSY) {
-                return (
-                  time < timeMinute.offset(block.end_time, event_buffer) &&
-                  timeMinute.offset(block.start_time, -event_buffer) < endTime
-                );
-              }
-            });
+            const slot = {
+              start_time: time,
+              end_time: endTime,
+              available_calendars: [],
+            };
+            const timeslots =
+              calendar.availability === AvailabilityStatus.BUSY
+                ? calendar.timeslots.map((t) => ({
+                    start_time: timeMinute.offset(t.start_time, -event_buffer),
+                    end_time: timeMinute.offset(t.end_time, event_buffer),
+                    available_calendars: [],
+                  }))
+                : calendar.timeslots;
+            const slotAvailability = overlap([...timeslots], slot);
```

Comment:
Does `timeslots` need to be spread here since it's already an array, and nothing is being added to it?